Mathlib/Data/Num/Lemmas.lean

```diff
@@ -1091,21 +1091,19 @@ theorem ofZNum'_toNat : ∀ n : ZNum, (↑) <$> ofZNum' n = Int.toNat' n
       show ((p.pred' + 1 : ℕ) : ℤ) = p by rw [← succ'_to_nat]; simp
 
 @[simp]
-theorem ofZNum_toNat : ∀ n : ZNum, (ofZNum n : ℕ) = Int.toNat n
```

Comment:
Please add a deprecation for this lemma, making it an alias of `cast_ofZNum`.

Reply:
On revisiting this, I realized that the approach I should have originally taken is not getting rid of cast_ofZNum since it's a useful helper lemma, but rather just removing @[simp] to satisfy the simpNF linter, so that's what I've done instead.